### DISCUSSION TITLE
BatchSpanProcessor.forceFlush() doesn't appear to flush anything

### DISCUSSION THREAD

**Discussion by kenfinnigan** [open-telemetry/opentelemetry-java#2805] [Q&A]
I was looking at shutdown hooks and following the code and I happened to notice that https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java#L247 doesn't appear to actually flush any spans that might be present.

Am I missing where it happens or is it a bug?

**Comment by jkwatson** [comment]
This looks like a bug, for sure. Needs to flush the exporter when called, and use that as the result code.

**Reply by jkwatson** [reply]
I take it back... there's actually some subtlety in this implementation. The worker thread checks for a request for flushing every 100ms and will initiate the flush. So `forceFlush()` just sets this value and the worker thread picks it up and uses it. I think not a bug, actually. The code does need a comment about this, though! I will put in a PR to add it.

**Reply by jkwatson** [reply]
https://github.com/open-telemetry/opentelemetry-java/pull/2807

**Reply by kenfinnigan** [reply]
Thanks!
